Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormats.cpp

```diff
@@ -53,24 +53,29 @@ struct DMABufFormat {
 };
 
 struct DMABufFormatsGroup {
-    DMABufFormatsGroup(const char* device, WPEBufferDMABufFormatUsage usage)
-        : device(device)
+    DMABufFormatsGroup(WPEDRMDevice* device, WPEBufferDMABufFormatUsage usage)
+        : device(device ? wpe_drm_device_ref(device) : nullptr)
         , usage(usage)
     {
     }
 
-    ~DMABufFormatsGroup() = default;
+    ~DMABufFormatsGroup()
+    {
+        g_clear_pointer(&device, wpe_drm_device_unref);
+    }
+
     DMABufFormatsGroup(const DMABufFormatsGroup&) = delete;
     DMABufFormatsGroup& operator=(const DMABufFormatsGroup&) = delete;
     DMABufFormatsGroup(DMABufFormatsGroup&& other)
-        : device(WTFMove(other.device))
+        : device(other.device)
         , usage(other.usage)
         , formats(WTFMove(other.formats))
     {
+        other.device = nullptr;
         other.usage = WPE_BUFFER_DMA_BUF_FORMAT_USAGE_RENDERING;
     }
 
-    CString device;
+    WPEDRMDevice* device { nullptr };
```

Comment:
Could we add a `GRefPtrWPEPlatform.{h,cpp}` with the template specialization to use `GRefPtr<WPEDRMDevice>`? Then it could be used here, simplifying the move constructor, and the destructor could be left untouched (and let the compiler generate it).

Reply:
Yes, for WPEDRMDevice and others I guess, so maybe it would be better in a follow up.